### ISSUE TITLE
single_component_path_imports should respect #[allow(unused_imports)]

### ISSUE THREAD

**Issue by dhardy** [rust-lang/rust-clippy#9139]
### Summary

To achieve dynamic linking, I use this:
```rust
#[cfg(feature = "dynamic")]
#[allow(unused_imports)]
use kas_dylib;
```
Without this import, built executables do not dynamically link this library.
Bevy [does the same](https://github.com/bevyengine/bevy/blob/main/src/lib.rs#L48).

Clippy complains:
```
warning: this import is redundant
  --> src/lib.rs:59:1
   |
59 | use kas_dylib;
   | ^^^^^^^^^^^^^^ help: remove it entirely
   |
   = note: `#[warn(clippy::single_component_path_imports)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
```

Since this import already has `#[allow(unused_imports)]` I would expect Clippy to ignore it.

### Lint Name

single_component_path_imports

### Version

```text
rustc 1.64.0-nightly (2f3ddd9f5 2022-06-27)
```

**Comment by alex-semenyuk** [comment]
@dhardy Is it still issue?
Can't reproduce it [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b614bb8c2525c3e91fd45388d585996e)

**Comment by dhardy** [comment]
Yes. On [kas master](https://github.com/kas-gui/kas/commit/6a99b5fe8dcdf1385b091dad00fc97e98d0556e3):
```
❯ rustc +nightly -V
rustc 1.64.0-nightly (f6f9d5e73 2022-08-04)
❯ cargo +nightly clippy --all-features
[...]
    Checking kas v0.11.0 (/home/dhardy/projects/kas/kas)
warning: this import is redundant
   --> src/lib.rs:112:1
    |
112 | use kas_dylib;
    | ^^^^^^^^^^^^^^ help: remove it entirely
    |
    = note: `#[warn(clippy::single_component_path_imports)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
```

**Comment by Ahmard** [comment]
Clippy now respects #[allow(unused_imports)], this issue can now be closed.

**Comment by dhardy** [comment]
... and now it doesn't again:
```
warning: this import is redundant
  --> src/lib.rs:69:1
   |
69 | use kas_dylib;
   | ^^^^^^^^^^^^^^ help: remove it entirely
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
   = note: `#[warn(clippy::single_component_path_imports)]` on by default
```

```
$ cargo clippy --version
clippy 0.1.90 (b56aaec52b 2025-07-24)
```

Edit: the same applies to `clippy 0.1.91 (898aff704d 2025-08-14)`.
